src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,105 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
- * реалізації ArrayList (default capacity, newCapacity...)</p>
- */
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] storage;
+    private int size;
+
+    public ArrayList() {
+        this.storage = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        checkCapacity();
+        storage[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+        } else {
+            if (index < 0 || index > size) {
+                throw new ArrayIndexOutOfBoundsException("Element with such index doesn't exist!");
+            }
```

Comment:
you can use your `indexCheck()` here
you have already covered the case when `index == size`